samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -111,27 +112,35 @@ public LocalApplicationRunner(SamzaApplication app, Config config) {
    * @param metadataStoreFactory the instance of {@link MetadataStoreFactory} to read and write to coordinator stream.
    */
   public LocalApplicationRunner(SamzaApplication app, Config config, MetadataStoreFactory metadataStoreFactory) {
-    this(ApplicationDescriptorUtil.getAppDescriptor(app, config), getCoordinationUtils(config), metadataStoreFactory);
+    this(app, config, getCoordinationUtils(config), metadataStoreFactory);
   }
 
   /**
    * Constructor only used in unit test to allow injection of {@link LocalJobPlanner}
    */
   @VisibleForTesting
-  LocalApplicationRunner(ApplicationDescriptorImpl<? extends ApplicationDescriptor> appDesc, Optional<CoordinationUtils> coordinationUtils) {
-    this(appDesc, coordinationUtils, getDefaultCoordinatorStreamStoreFactory(new JobConfig(appDesc.getConfig())));
+  LocalApplicationRunner(SamzaApplication app, Config config, Optional<CoordinationUtils> coordinationUtils) {
+    this(app, config, coordinationUtils, getDefaultCoordinatorStreamStoreFactory(new JobConfig(config)));
   }
 
   private LocalApplicationRunner(
```

Comment:
Is this considered a semantic API change? Before, it seems like `config` was always the full config, but now you are allowing the initial config with only config loader properties.
It's ok to have this be an API change, but just wanted to make sure you note that where necessary.

Reply:
Good question, I would consider changes in samza-api module to be public facing API changes, and changes in samza-core to be internal API changes, which can be less risky. 

I will update the PR description to explicitly call out the constructor changes, also this is a backward compatible API change.

Reply:
`LocalApplicationRunner` is kind of in that middle layer: it's not in samza-api and not quite a direct API, but it also can get used by users running standalone and it can be specified in configs as the "app.runner.class".

Reply:
In my opinion, `LocalApplicationRunner` doesn't truly belong in samza-core, but it had to be somewhere...

Reply:
Got you, in addition to note this change in PR's description, anywhere else we should notify this change?

Reply:
Someone will need to remember to add this as a release note, and if we have information in the user guide about this flow, then that will need to be updated.

Reply:
I will make sure this to be added in the OSS release when it is out and I will go through our user guide to see if we have any thing about this flow.